src/edu/ucsd/library/xdre/web/CollectionStatusReportController.java

```diff
@@ -0,0 +1,260 @@
+
+                    String fileName = "collection_report" + beginDate.replace("-", "") + ".csv";
+                    OutputStream out = response.getOutputStream();
+                    response.setHeader("Content-Disposition", "inline; filename=" + fileName);
+                    response.setContentType("text/csv");
+                    out.write(reportBuilder.toString().getBytes());
+                    out.close();
+                    return null;
+                } else {
+                    dataMap.put("recordReleased", collectionsReleased);
+                    dataMap.put("recordAdded", collectionsAdded);
+                    dataMap.put("recordRemoved", collectionsRemoved);
+                    dataMap.put("recordEdited", collectionsEdited);
+                    dataMap.put("beginDate", beginDate);
+                    dataMap.put("endDate", endDate);
+                }
+            } catch (Exception e){
+                e.printStackTrace();
+                dataMap.put("message", "Error: " + e.getMessage());
+            } finally {
+                if (damsClient != null)
+                    damsClient.close();
+                if (damsClientEvent != null)
+                    damsClientEvent.close();
+            }
+        } else {
+            dataMap.put("recordReleased", new HashMap<String, String>());
+            dataMap.put("recordAdded", new HashMap<String, String>());
+            dataMap.put("recordRemoved", new HashMap<String, String>());
+            dataMap.put("recordEdited", new HashMap<String, String>());
+            dataMap.put("beginDate", "");
+            dataMap.put("endDate", "");
+        }
+        return new ModelAndView("collectionStatusReport", "model", dataMap);
+    }
+
+    public static List<String> getValues(Document doc, String xPath){
+        List<String> values = new ArrayList<String>();
+        List<Node> nodes = doc.selectNodes(xPath);
+        for(Iterator<Node> it=nodes.iterator(); it.hasNext();)
+            values.add(it.next().getText());
+        return values;
+    }
+
+    /* 
+     * Build SPARQL for events:
+     * predicate dams:type http://library.ucsd.edu/ark:/20775/bd3106617w
+     * predicate dams:eventDate http://library.ucsd.edu/ark:/20775/bd5120287c
+     * @param eventType
+     * @param beginDate
+     * @return
+     */
+    private static String buildEventSparql(String eventType, String beginDate, String endDate) throws ParseException {
+        String searchEndDate = endDate;
+        if (StringUtils.isNotBlank(endDate)) {
+            // set date range to include the whole end date
+            Calendar eCal = Calendar.getInstance();
+            eCal.setTime(dateFormat.parse(endDate));
+            eCal.add(Calendar.DATE, 1);
+            searchEndDate = dateFormat.format(eCal.getTime());
+        }
+
+        String sparql = "SELECT ?e WHERE {?e <http://library.ucsd.edu/ark:/20775/bd3106617w> '\""
```

Comment:
This is just an idea, would it be worth making the Arks used in this and the other queries constants with names that refer to their purpose/role? Just an idea.